src/diamonds/nayms/libs/LibEntity.sol

```diff
@@ -163,8 +163,21 @@ library LibEntity {
         AppStorage storage s = LibAppStorage.diamondStorage();
         validateEntity(_entity);
 
+        uint256 oldCollateralRatio = s.entities[_entityId].collateralRatio;
+        uint256 oldUtilizedCapacity = s.entities[_entityId].utilizedCapacity;
+
         s.entities[_entityId] = _entity;
 
+        // if it's a cell, and collateral ratio changed
+        if (_entity.assetId != 0 && _entity.collateralRatio != oldCollateralRatio) {
+            // unlock current utilized capacity
+            s.lockedBalances[_entityId][_entity.assetId] -= oldUtilizedCapacity;
+            // recalculate utilized capacity
+            s.entities[_entityId].utilizedCapacity = (oldUtilizedCapacity * _entity.collateralRatio) / oldCollateralRatio;
+            // lock updated utilized capacity
+            s.lockedBalances[_entityId][_entity.assetId] += s.entities[_entityId].utilizedCapacity;
+        }
+
         emit EntityUpdated(_entityId);
```

Comment:
Perhaps it'll be useful to emit the previous and new utilized capacity here?

Reply:
it does not necessarily change, I guess inside the if statement we could, emit a dedicated event i.e. `CollateralRatioUpdated` with relevant values 🤔